### PR TITLE
Defaulting logging env var. Added logging samples

### PR DESCRIPTION
**Pull request by jeremydvoss**
* Defaulting _OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED to true for auto-instrumentation.
 * Added samples for correlated logs, exceptions, and custom properties
 * Updated readme with logging_level defaulting behavior.
Fixes https://github.com/microsoft/ApplicationInsights-Python/issues/239

### CODE REVIEW

File: azure-monitor-opentelemetry-distro/samples/logging/custom_properties.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    logger_name=__name__,
+    logging_level=DEBUG,
+    disable_metrics=True,
+    disable_tracing=True,
+)
+
+logger = getLogger(__name__)
+logger.setLevel(DEBUG)
+
+logger.debug("DEBUG: Debug with properties", extra={"debug": "true"})
```
Comment (lzchen) [inline azure-monitor-opentelemetry-distro/samples/logging/custom_properties.py:23]: Can you add a comment on what the format of "extra" should be?

File: azure-monitor-opentelemetry-distro/samples/logging/correlated_logs.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    connection_string="<your-connection-string>",
+    service_name="foo_service",
+    logging_level=WARN,
+    disable_metrics=True,
+    disable_tracing=True,
+)
+
+logger = getLogger(__name__)
+tracer = trace.get_tracer(__name__)
+
```
Comment (lzchen) [inline azure-monitor-opentelemetry-distro/samples/logging/correlated_logs.py:22]: Should probably have a log outside of the span to show the difference.

File: azure-monitor-opentelemetry-distro/samples/logging/correlated_logs.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from logging import WARN, getLogger
```
Comment (lzchen) [inline azure-monitor-opentelemetry-distro/samples/logging/correlated_logs.py:7]: I think `WARN` is actually deprecated so we should be using `logging.WARNING`.

File: azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/distro.py [hunk 4]
```diff
@@ -59,6 +62,9 @@ def _configure_auto_instrumentation() -> None:
         environ.setdefault(
             OTEL_LOGS_EXPORTER, "azure_monitor_opentelemetry_exporter"
         )
+        environ.setdefault(
```
[the remark this answers is not in the document]
Comment (jeremydvoss) [reply azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/distro.py:65]: No, it's just for auto-instrumentation:
https://github.com/open-telemetry/opentelemetry-python/blob/209093bfbe3fae0fc29c693f55d6bb6a2d997aad/opentelemetry-sdk/src/opentelemetry/sdk/_configuration/__init__.py#L353

File: azure-monitor-opentelemetry-distro/samples/logging/custom_properties.py (hunk 1 above)
Comment (jeremydvoss) [reply azure-monitor-opentelemetry-distro/samples/logging/custom_properties.py:23]: Could you be more specific? Like just a comment saying "pass custom properties as a dictionary with the extra field"?

File: azure-monitor-opentelemetry-distro/samples/logging/correlated_logs.py (hunk 2 above)
Comment (jeremydvoss) [reply azure-monitor-opentelemetry-distro/samples/logging/correlated_logs.py:22]: Done

File: azure-monitor-opentelemetry-distro/samples/logging/correlated_logs.py (hunk 3 above)
Comment (jeremydvoss) [reply azure-monitor-opentelemetry-distro/samples/logging/correlated_logs.py]: Done.

**Review by lzchen — APPROVED** [review]

**Pull request merged**